resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/submitsteps/hadoopsteps/HadoopConfMounterStep.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.submit.submitsteps.hadoopsteps
+
+import java.io.File
+
+import org.apache.commons.io.FileUtils.readFileToString
+
+import org.apache.spark.deploy.k8s.{HadoopConfBootstrap, PodWithMainContainer}
+import org.apache.spark.deploy.k8s.constants._
+
+ /**
+  * This step is responsible for taking the contents from each file in
+  * HADOOP_CONF_DIR, grabbing its contents as a string and storing each of them
+  * as a key-value pair in a configmap. Each key-value pair will be stored
+  * as a file, via Volume Mounts, later. The HADOOP_CONF_DIR_LOC is passed into the
+  * SchedulerBackend via sparkConf.
+  */
+private[spark] class HadoopConfMounterStep(
+    hadoopConfigMapName: String,
+    hadoopConfigurationFiles: Seq[File],
+    hadoopConfBootstrapConf: HadoopConfBootstrap,
+    hadoopConfDir: String)
+  extends HadoopConfigurationStep {
+
+   override def configureContainers(hadoopConfigSpec: HadoopConfigSpec): HadoopConfigSpec = {
+    val bootstrappedPodAndMainContainer =
+      hadoopConfBootstrapConf.bootstrapMainContainerAndVolumes(
+        PodWithMainContainer(
+          hadoopConfigSpec.driverPod,
+          hadoopConfigSpec.driverContainer))
+     hadoopConfigSpec.copy(
+       driverPod = bootstrappedPodAndMainContainer.pod,
+       driverContainer = bootstrappedPodAndMainContainer.mainContainer,
+       configMapProperties =
+         hadoopConfigurationFiles.map(file =>
+           (file.toPath.getFileName.toString, readFileToString(file))).toMap,
```

Comment:
Does the encoding of the strings from the files matter?

Reply:
No it doesn't. I thought this method was the most simple

Reply:
Looks like this version of `readFileToString ` has been deprecated. https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#readFileToString(java.io.File). I would suggest using https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#readFileToString(java.io.File,%20java.lang.String) with `UTF-8`.

Reply:
Or you can use Guava's `Files.toString` https://google.github.io/guava/releases/19.0/api/docs/com/google/common/io/Files.html#toString(java.io.File, java.nio.charset.Charset) with https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Charsets.html#UTF_8.

Reply:
Always hard encode to UTF-8. It's unclear between JVMs if the default encoding will be consistent. +1 for using Guava's `Files.toString`. I think we read files into string contents elsewhere in the codebase - be consistent with those.